tests/yam/agama/agama.pm

```diff
@@ -0,0 +1,54 @@
+## Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Run interactive installation with Agama,
+# run playwright tests directly from the Live ISO.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base Yam::agama::agama_base;
+use strict;
+use warnings;
+
+use testapi qw(
+  assert_screen
+  assert_script_run
+  select_console
+  get_required_var
+  upload_logs
+  match_has_tag
+  reset_consoles
+);
+use power_action_utils 'power_action';
+use transactional 'process_reboot';
+
+sub run {
+    assert_screen(['agama-main-page', 'agama_product_selection'], 120);
+    $testapi::password = 'linux';
+    select_console 'root-console';
+
+    my $type = get_required_var('AGAMA_TEST');
+    my $cmd;
+    if ($type !~ /take_screenshots/) {
+        $cmd = 'PRODUCTNAME="' . get_required_var('AGAMA_PRODUCT') . '"';
+    } else {
+        $cmd = 'RUN_INSTALLATION=1';
+    }
+    assert_script_run($cmd . ' playwright test --trace on --project chromium --config /usr/share/e2e-agama-playwright ' . $type, timeout => 1200);
+
+    $testapi::password = 'nots3cr3t' if ($type =~ /encrypted_lvm|full-disk-encryption/);
+    assert_script_run('reboot');
+    # For agama test, it is too short time to match the grub2, so we create
+    # a new needle to avoid too much needles loaded.
+    if ($type =~ /take_screenshots|lvm/) {
```

Comment:
this part here should be more "lineal" and covers both scenarios, encrypted and not encrypted.